test/terra/states/test_aer_statevector.py

```diff
@@ -0,0 +1,1154 @@
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Integration Tests for AerStatevector
+"""
+
+import unittest
+from math import pi
+import numpy as np
+import logging
+from itertools import permutations
+from ddt import ddt, data
+from numpy.testing import assert_allclose
+
+from qiskit import transpile
+from qiskit.exceptions import QiskitError
+from qiskit.circuit import QuantumCircuit, QuantumRegister
+from qiskit.providers.basicaer import QasmSimulatorPy
+from qiskit.quantum_info.random import random_unitary, random_statevector, random_pauli
+from qiskit.quantum_info.states import Statevector
+from qiskit.circuit.library import QuantumVolume
+from qiskit.providers.aer import AerSimulator
+from qiskit.quantum_info.operators.operator import Operator
+from qiskit.quantum_info.operators.symplectic import Pauli, SparsePauliOp
+from qiskit.quantum_info.operators.predicates import matrix_equal
+from qiskit.visualization.state_visualization import numbers_to_latex_terms, state_to_latex
+from qiskit.circuit.library import QFT, HGate
+from test.terra import common
+from qiskit_aer.aererror import AerError
+from qiskit_aer.quantum_info.states import AerStatevector
+
+
+logger = logging.getLogger(__name__)
+
+@ddt
+class TestAerStatevector(common.QiskitAerTestCase):
+    """AerState tests"""
+
+    def test_generate_aer_statevector(self):
+        """Test generation of Aer's StateVector"""
+        
+        circ = QuantumCircuit(5)
+        state = AerStatevector(circ)
+
+    def test_qv(self):
+        """Test generation of Aer's StateVector with QV """
+        circ = QuantumVolume(5, seed=1111)
+        state = AerStatevector(circ)
+        expected = Statevector(circ)
+
+        for e, s in zip(expected, state):
+            self.assertAlmostEqual(e, s)
+
+    def test_method_and_device_properties(self):
+        """Test method and device properties"""
+        circ = QuantumVolume(5, seed=1111)
+        state1 = AerStatevector(circ)
+
+        self.assertEqual('statevector', state1.metadata['method'])
+        self.assertEqual('CPU', state1.metadata['device'])
+
+        state2 = AerStatevector(circ, method='matrix_product_state')
+        self.assertEqual('matrix_product_state', state2.metadata['method'])
+        self.assertEqual('CPU', state2.metadata['device'])
+
+        for pa1, pa2 in zip(state1, state2):
+            self.assertAlmostEqual(pa1, pa2)
+
+    def test_evolve(self):
+        """Test method and device properties"""
+        circ1 = QuantumVolume(5, seed=1111)
+        circ2 = circ1.compose(circ1)
+        circ3 = circ2.compose(circ1)
+
+        state1 = AerStatevector(circ1)
+        state2 = AerStatevector(circ2)
+        state3 = AerStatevector(circ3)
+
+        for pa1, pa2 in zip(state1.evolve(circ1), state2):
+            self.assertAlmostEqual(pa1, pa2)
```

Comment:
I think this is largely equivalent to `self.assertEqual(state1.evolve(circ1), state2))`, since the equality comparison in `Statevector` is defined to be fuzzy (though that certainly wouldn't have been my choice...).